src/Pos/Block/Logic.hs

```diff
@@ -905,14 +926,13 @@ createMainBlockPure limit prevHeader txs pSk sId psks sscData usPayload sk =
                 usPayload' <- includeUSPayload
                 return (psks', usPayload')
         -- include transactions
-        txs' <- takeSome (map snd txs)
+        txs' <- takeSome txs
         -- return the resulting block
         txPayload <- either throwError pure $ mkTxPayload txs'
-        let body = Types.MainBody txPayload sscData psks' usPayload'
-        maybe (error "Coudln't create block") return $
-              mkMainBlock (Just prevHeader) sId sk pSk body extraH extraB
+        let body = Types.MainBody txPayload sscPayload psks' usPayload'
+        let finalBlock = mkMainBlock (Just prevHeader) sId sk pSk body extraH extraB
+        either (\s -> throwError $ "Couldn't create block: " <> s) pure finalBlock
```

Comment:
Thanks for getting rid of that `error`. Also I would generalize type of `mkMainBlock` to `MonadError`, it would eliminate some `either`s.